doc/developer/design/20231113_optimizer_notice_catalog.md

```diff
@@ -0,0 +1,480 @@
+  Action_ statement which contains concrete instructions that will resolve the
+  issue. If possible, the corrective action is given as a concrete SQL snippet,
+  otherwise it is a valid markdown string.
+- Every detailed view has a "Learn more" link below the notice section that
+  references the public docs for this notice type. The notice docs are
+  structured similar to [the Rust error codes page][rust_error_codes] and will
+  provide more context, a minimal reproducing example, and a corrective action
+  for the example for each notice type for customers that want to understand
+  where and why a specific notice is emitted.
+
+### Catalog Schema
+
+#### <a id="mz_optimizer_notices"></a> `mz_internal.mz_optimizer_notices` (new)
+
+Optimizer notices are exposed through the `mz_internal.mz_optimizer_notices`
+view with the following columns.
+
+| Field                   | Type                         | Meaning                                                                                                                                           |
+| ----------------------- | ---------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `notice_type`           | `text`                       | The notice type. Each type will correspond to a Materialize docs sub-page.                                                                        |
+| `message`               | `text`                       | A brief description of the issue highlighted by this notice.                                                                                      |
+| `hint`                  | `text`                       | A high-level hint that tells the user what can be improved.                                                                                       |
+| `action`                | `text`                       | A concrete action that will resolve the notice.                                                                                                   |
+| `redacted_message`      | `text`                       | A redacted version of the `message` column. `NULL` if no redaction is needed.                                                                     |
+| `redacted_hint`         | `text`                       | A redacted version of the `hint` column. `NULL` if no redaction is needed.                                                                        |
+| `redacted_action`       | `text`                       | A redacted version of the `action` column. `NULL` if no redaction is needed.                                                                      |
+| `action_type`           | `text`                       | The type of the `action` string (`sql_statements` for a valid SQL string or `plain_text` for plain text).                                         |
```

Comment:
We are recording `action_type` because at some point in the future we might want to add a button that simply executes it if it is `sql_statements`?

Reply:
Yes (this request came from @sthm). A button probably won't cut it until we have zero-downtime `REPLAN` which seems some way down the road. However, I guess even now it is better UX if the customer can copy-paste a valid SQL string the notice into their dbt project or whatever else they use to populate their catalog.